src/connection.js

```diff
@@ -591,43 +615,106 @@ export class Connection {
   /**
    * Fetch the balance for the specified public key
    */
-  async getBalance(
+  async getBalanceAndContext(
     publicKey: PublicKey,
     commitment: ?Commitment,
-  ): Promise<number> {
+  ): Promise<RpcResponseAndContext<number>> {
     const args = this._argsWithCommitment([publicKey.toBase58()], commitment);
     const unsafeRes = await this._rpcRequest('getBalance', args);
-    const res = GetBalanceRpcResult(unsafeRes);
+    const res = GetBalanceAndContextRpcResult(unsafeRes);
     if (res.error) {
       throw new Error(res.error.message);
     }
     assert(typeof res.result !== 'undefined');
-    return res.result;
+
+    const isV21 =
+      typeof res.result.context !== 'undefined' &&
+      typeof res.result.value !== 'undefined';
+
+    const slot = isV21 ? res.result.context.slot : NaN;
+    let value = isV21 ? res.result.value : res.result;
+
+    const rpcResponseAndContext = {
+      context: {
+        slot,
+      },
+      value,
+    };
+
+    return rpcResponseAndContext;
+  }
+  async getBalance(
+    publicKey: PublicKey,
+    commitment: ?Commitment,
+  ): Promise<number> {
+    return await this.getBalanceAndContext(publicKey, commitment)
+      .then(x => x.value)
+      .catch(e => {
+        throw e;
+      });
   }
 
   /**
    * Fetch all the account info for the specified public key
    */
-  async getAccountInfo(
+  async getAccountInfoAndContext(
     publicKey: PublicKey,
     commitment: ?Commitment,
-  ): Promise<AccountInfo> {
+  ): Promise<RpcResponseAndContext<AccountInfo>> {
     const args = this._argsWithCommitment([publicKey.toBase58()], commitment);
     const unsafeRes = await this._rpcRequest('getAccountInfo', args);
-    const res = GetAccountInfoRpcResult(unsafeRes);
+    const res = GetAccountInfoAndContextRpcResult(unsafeRes);
     if (res.error) {
       throw new Error(res.error.message);
     }
+    assert(typeof res.result !== 'undefined');
 
-    const {result} = res;
-    assert(typeof result !== 'undefined');
+    const isV21 =
+      typeof res.result.context !== 'undefined' &&
+      typeof res.result.value !== 'undefined';
+
+    const slot = isV21 ? res.result.context.slot : NaN;
+    let value = null;
+    if (isV21 && res.result.value) {
+      const {executable, owner, lamports, data} = res.result.value;
+      value = {
+        executable,
+        owner: new PublicKey(owner),
+        lamports,
+        data: Buffer.from(data),
+      };
+    } else if (!isV21 && res.result) {
+      const {executable, owner, lamports, data} = res.result;
+      value = {
+        executable,
+        owner: new PublicKey(owner),
+        lamports,
+        data: Buffer.from(data),
+      };
+    }
```

Comment:
How about:
```js
    let resultValue = isV021 ? res.result.value : res.result;
    assert(typeof resultValue !== 'undefined');
    const {executable, owner, lamports, data} = resultValue;
    let value = {
        executable,
        owner: new PublicKey(owner),
        lamports,
        data: Buffer.from(data),
    };
```

Reply:
resultValue can be null, would that destructure ok? is there a slight tweak for that? 😕

Reply:
ah ok just realized the reorder caught that! thanks!